cmake/third_party/aws_sdk.cmake

```diff
@@ -0,0 +1,34 @@
+include(ExternalProject)
+include(GNUInstallDirs)
+
+# Download and install the AWS CPP SDK into the build directory.
+ExternalProject_Add(aws-sdk
+    PREFIX aws-sdk-cpp
+    GIT_REPOSITORY https://github.com/aws/aws-sdk-cpp.git
+    GIT_TAG 1.9.175
+    CMAKE_ARGS       
+        -DBUILD_SHARED_LIBS=ON
+        -DBUILD_ONLY=s3-crt
+        -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install
+        -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+    BUILD_ALWAYS FALSE
+    INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install
+    BUILD_BYPRODUCTS
+        ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX}
+        ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}
+    TEST_COMMAND      ""
+    UPDATE_COMMAND    ""
+)
+ExternalProject_Get_Property(aws-sdk INSTALL_DIR)
+
+add_library(aws-sdk::core SHARED IMPORTED)
+add_library(aws-sdk::s3-crt SHARED IMPORTED)
+add_library(aws-sdk::crt SHARED IMPORTED)
+set_target_properties(aws-sdk::core PROPERTIES
+    IMPORTED_LOCATION ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}                                             
```

Comment:
I'd append a second property to this call `INTERFACE_INCLUDE_DIRECTORIES` to point to the installed includes for the aws-sdk i.e.
```
INTERFACE_INCLUDE_DIRECTORIES ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR}
```
This means that anything that links against `aws-sdk::core` will automatically add `{INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR}` to its include path.

Reply:
I get a compiler error when I add this line -    ` {INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR} `translates to `"/home/ubuntu/wiredtiger/build/aws-sdk-cpp/install/include"` and this path doesn't exist

Reply:
Ah right, we probably need to pre-define the include directory for `INTERFACE_INCLUDE_DIRECTORIES` to make CMake happy at the configuration stage. You could add the following before this line:
```
 file(MAKE_DIRECTORY {INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR})
```